File: MP.UnmanagedCode/MP.UnmanagedCode.PowerManagementAPI.Tests/HibernationFileManagerTests.cs
using Xunit;

namespace MP.UnmanagedCode.PowerManagementAPI.Tests
{
    public class HibernationFileManagerTests
    {
        private readonly HibernationFileManager _hibernationFileManager;

        public HibernationFileManagerTests()
        {
            _hibernationFileManager = new HibernationFileManager();
        }

        [Fact]
        public void ReserveHibernationFile_Test()
        {
            _hibernationFileManager.ReserveHibernationFile();
        }

        [Fact]
        public void DeleteHibernationFile_Test()
        {
            _hibernationFileManager.DeleteHibernationFile();
        }

        [Fact (Skip = "Switches to hibernation mode")]
        public void SwitchToHibernationModeToHibernationMode_DisableWakeEventsFalse_Test()
        {
            _hibernationFileManager.SwitchToHibernationMode();
        }

        [Fact(Skip = "Switches to hibernation mode")]
        public void SwitchToHibernationModeToHibernationMode_DisableWakeEventsTrue_Test()
        {
            _hibernationFileManager.SwitchToHibernationMode(disableWakeEvents: true);
        }
    }
}

File: MP.UnmanagedCode/MP.UnmanagedCode.PowerManagementAPI.VBTests/HibernationFileManagerTests.vb
Imports Xunit

Public Class HibernationFileManagerTests
    Private ReadOnly _hibernationFileManager As HibernationFileManager

    Sub New()
        _hibernationFileManager = New HibernationFileManager()
    End Sub

    <Fact>
    Sub ReserveHibernationFile_Test()
        _hibernationFileManager.ReserveHibernationFile()
    End Sub

    <Fact>
    Sub DeleteHibernationFile_Test()
        _hibernationFileManager.DeleteHibernationFile()
    End Sub

    <Fact(Skip := "Switches to hibernation mode")>
    Sub SwitchToHibernationModeToHibernationMode_DisableWakeEventsFalse_Test()
        _hibernationFileManager.SwitchToHibernationMode()
    End Sub

    <Fact(Skip := "Switches to hibernation mode")>
    Sub SwitchToHibernationModeToHibernationMode_DisableWakeEventsTrue_Test()
        _hibernationFileManager.SwitchToHibernationMode(disableWakeEvents := true)
    End Sub

End Class